src/chat/ChatScreen.js

```diff
@@ -24,7 +24,7 @@ import { getShownMessagesForNarrow, isNarrowValid as getIsNarrowValid } from './
 
 type Props = $ReadOnly<{|
   navigation: AppNavigationProp<'chat'>,
-  route: RouteProp<'chat', {| narrow: Narrow |}>,
+  route: RouteProp<'chat', {| narrow: Narrow, editMessage: EditMessage | null |}>,
```

Comment:
```
ChatScreen: Store `editMessage` as a nav param, not React state.

The fact that `ChatScreen`'s header changes with this bit of state
is a clue that it really wants to be a route param. When we start
using the `header` option [1] for the header, instead of smushing
the header in with the rest of the screen's content, it'll be
convenient to grab `editMessage` from the route params.

[1] https://reactnavigation.org/docs/stack-navigator/#header
```

I think we should aim to reduce the contents of `editMessage`, probably to just the ID of the message being edited, along the lines of [what React Navigation recommends should go in these params](https://reactnavigation.org/docs/params/#what-should-be-in-params).

I think the biggest drawback to storing the whole original content and topic equally applies in the status quo (i.e., where `editMessage` is stored in React state): it's model data, so it can become stale. So, perhaps a fix can be done as a followup, or we can add a TODO.

The article gives a couple of other good reasons, but some of them (like gracefully handling deep-linking) don't yet really apply to us. But I think their guidance that "You can also think of the route object like a URL" makes sense. For this screen, I'm thinking of the analogy of a URL like `/msglist?msg-edit-id=12345`.

Reply:
In my draft that moves `ModalSearchNavBar` to the header, the search term is stored in the nav state. This seems fine; I'm thinking of the analogy of a URL that has something like `?q=foo`.

Reply:
> In my draft that moves `ModalSearchNavBar` to the header

While at first I thought we might not want the search bar to go in the header/nav bar, I think it's a fine place to put it, at least on Android.

From an [Android doc](https://developer.android.com/training/appbar):

> The key functions of the app bar are as follows:
> - [...]
> - Access to important actions in a predictable way, such as search.

Reply:
> I think the thing that makes that tricky is mainly just the message content.

Ahh—right, you're right. I saw that but forgot about it when commenting.